packages/web/src/scss/tools/_utilities.scss

```diff
@@ -1,13 +1,56 @@
-// Inspired by React UI and Bootstrap
+// 1. Format key if it's type of number so it's always 2 digits
+// 2. Don't prefix if value key is null (e.g. with shadow class)
+// 3. Remove default postfix from class name
+
+@use '@tokens' as tokens;
 @use 'sass:list';
 @use 'sass:map';
 @use 'sass:meta';
-@use 'string';
+@use 'sass:string';
+@use 'string' as spirit-string;
+@use 'tokens' as tokens-tools;
+
+// Generates a map of tokens based on the provided key, prefix, replacement, and scope.
+// It iterates over the tokens and creates a new map with modified keys and values.
+// Parameters are:
+// * $tokens-key: the key used to retrieve a group of tokens from `@tokens`.
+// * $prefix: a prefix to prepend to each key in the generated map.
+// * $replacement: a string to be replaced in the token keys.
+// * $scope: a list of keys to filter tokens included in the generated map.
+// Returns a map of tokens with modified keys and values.
+@function generate-token-map($tokens-key: null, $prefix: null, $replacement: null, $scope: null) {
+    $result: ();
+    $tokens: tokens-tools.get-variable-if-exists($tokens-key, null);
+
+    @each $key, $values in $tokens {
+        @each $subkey, $value in $values {
+            @if meta.type-of($scope) == 'list' and $subkey != null and list.index($scope, $subkey) != null {
+                // 1.
+                $name: $prefix +
+                    if(meta.type-of($key) == 'number' and $key < 10, '0' + $key, $key) +
+                    '-' +
+                    spirit-string.replace($subkey, $replacement + '-', '');
+
+                $result: map.set($result, string.unquote($name), $value);
+            }
+        }
+    }
+
+    @return $result;
+}
 
+// Dynamically generates utility classes based on a given utility map.
+// It iterates over the values defined in the utility map and creates CSS classes with the specified properties.
+// Parameters are:
+// $utility - a map containing utility configuration. It should include:
+// * `values`: a map or list of key-value pairs for the utility values.
+// * `property`: a list of CSS properties to apply for each value.
+// * `class`: the base class name for the utility.
+// $infix - a string to append to the base class name, typically used for responsive or state-specific variations.
```

Comment:
Please unify the format of the parameters here (you are missing `*` before them).